src/components/Traceroute/Traceroute.tsx

```diff
@@ -55,89 +43,206 @@ const TracerouteTool = () => {
+     * @param {string} data - The data recieved from the child process.
+     */
+    const handleProcessData = useCallback((data: string) => {
+        setOutput((prevOutput) => prevOutput + "\n" + data); //Append new data to the previous output.
+    }, []);
+
+    /**
+     * handleProcessTermination: Callback to handle the termination of the child process.
+     * Once the process termination is handled, it clears the process PID reference and
+     * deactivates the loading overlay.
+     * @param {object} param - An object containing information about the process termination.
+     * @param {number} param.code - The exit code of the terminated process.
+     * @param {number} param.signal - The signal code indicating how the process was terminated.
+     */
+
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            // If the process was successful, display a success message.
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+
+                // If the process was terminated manually, display a termination message.
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+
+                // If the process was terminated with an error, display the exit and signal codes.
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+
+            // Clear the child process pid reference. There is no longer a valid process running.
+            setPid("");
+
+            // Cancel the loading overlay. The process has completed.
+            setLoading(false);
+        },
+        [handleProcessData] // Dependency on the handleProcessData callback
+    );
+
+    //Traceroute Options
+    const tracerouteSwitch = [
+        "Traceroute ICMP scan",
+        "Traceroute TCP scan",
+        "Traceroute UDP scan",
+        "Traceroute custom scan",
+    ]; // Options for different types of traceroute scans
+
+    /**
+     * onSubmit: Asynchronous handler for the form submission event event related to executing traceroute commands.
+      It constructs command arguments, executes them via a shell script, and updates the UI with the results or errors.
+     *
+     * @param {FormValuesType} values - Contains the type of traceroute scan (ICMP, TCP, UDP, custom),
+     * the destination hostname, and optional custom traceroute options.
+     */
     const onSubmit = async (values: FormValuesType) => {
         let args = [``];
 
-        //Switch case
+        // Switch case to handle different traceroute scan options based on user selection.
         switch (values.tracerouteSwitch) {
-            case "Traceroute ICMP scan": //traceroute syntax: traceroute -I <hostname>
+            case "Traceroute ICMP scan":
+                // Traceroute ICMP scan uses the '-I' option
```

Comment:
Move comments to before the case switch

Reply:
Moved the comments before the case switch